### ISSUE TITLE
QtCreator Segmentation fault

### ISSUE THREAD

**Issue by bialasjaroslaw** [msys2/MINGW-packages#6522]
I recently updated my environment to latest version

```
# qmake --version
QMake version 3.1
Using Qt version 5.14.2 in C:/msys2/mingw32/lib
```

Then I tried to run qtcreator and everything I got was `Segmentation fault`. First I tried to disable all plugins by `-noload all` option but without success. Then I tried to reinstall package and when that failed I did it with qt5 dev related stuff (qt5, qt-creator, etc). I run `qtcreator` in `gdb` and I got SEGFAULT in Qt5Core library 
```
Thread 1 received signal SIGSEGV, Segmentation fault.
0x75f7b122 in wcstok_s () from C:\WINDOWS\System32\msvcrt.dll
(gdb) bt
#0  0x75f7b122 in wcstok_s () from C:\WINDOWS\System32\msvcrt.dll
#1  0x75f7b063 in wcstok_s () from C:\WINDOWS\System32\msvcrt.dll
#2  0x669b9603 in QString::reallocData(unsigned int, bool) ()
   from C:\msys2\mingw32\bin\Qt5Core.dll
#3  0x669bf3dc in QString::utf16() const ()
   from C:\msys2\mingw32\bin\Qt5Core.dll
#4  0x6ded897c in QWindowsContext::registerWindowClass(QString, long ( stdcall*)(HWND__*, unsigned int, unsigned int, long), unsigned int, HBRUSH__*, bool)
    () from C:\msys2\mingw32\share\qt5\plugins\platforms\qwindows.dll
#5  0x6ded9709 in QWindowsContext::createDummyWindow(QString const&, wchar_t const*, long ( stdcall*)(HWND__*, unsigned int, unsigned int, long), unsigned long) () from C:\msys2\mingw32\share\qt5\plugins\platforms\qwindows.dll
#6  0x6df1f6e5 in QWindowsGLContext::QWindowsGLContext(QOpenGLStaticContext*, QOpenGLContext*) ()
   from C:\msys2\mingw32\share\qt5\plugins\platforms\qwindows.dll
#7  0x00000001 in ?? ()
#8  0x776238c0 in combase!CoFreeUnusedLibrariesEx ()
   from C:\WINDOWS\System32\combase.dll
#9  0x07acb6d0 in ?? ()
#10 0x6df6fb50 in QAccessibleInterface::interface_cast(QAccessible::InterfaceType) () from C:\msys2\mingw32\share\qt5\plugins\platforms\qwindows.dll
#11 0x6df227f0 in QWindowsGLContext::QWindowsGLContext(QOpenGLStaticContext*, QOpenGLContext*) ()
   from C:\msys2\mingw32\share\qt5\plugins\platforms\qwindows.dll
#12 0x6df83be0 in WindowCreationData::create(QWindow const*, QWindowsWindowData const&, QString) const ()
   from C:\msys2\mingw32\share\qt5\plugins\platforms\qwindows.dll
#13 0x6df71910 in QWindowsXpNativeFileDialog::~QWindowsXpNativeFileDialog() ()
   from C:\msys2\mingw32\share\qt5\plugins\platforms\qwindows.dll
#14 0x6df9e534 in _fu5___ZTVN10__cxxabiv117__class_type_infoE ()
   from C:\msys2\mingw32\share\qt5\plugins\platforms\qwindows.dll
#15 0x6df68400 in qt_addBitmapToPath(double, double, unsigned char const*, int, int, int, QPainterPath*) ()
   from C:\msys2\mingw32\share\qt5\plugins\platforms\qwindows.dll
#16 0x6df68420 in QThread::event(QEvent*) ()
   from C:\msys2\mingw32\share\qt5\plugins\platforms\qwindows.dll
#17 0x6df68428 in QThread::qt_metacast(char const*) ()
   from C:\msys2\mingw32\share\qt5\plugins\platforms\qwindows.dll
#18 0x6df6ff10 in QShGetFileInfoThread::~QShGetFileInfoThread() ()
   from C:\msys2\mingw32\share\qt5\plugins\platforms\qwindows.dll
#19 0x6df68458 in QObject::qt_metacall(QMetaObject::Call, int, void**) ()
   from C:\msys2\mingw32\share\qt5\plugins\platforms\qwindows.dll
#20 0x6df68468 in QObject::customEvent(QEvent*) ()
   from C:\msys2\mingw32\share\qt5\plugins\platforms\qwindows.dll
#21 0x6df68470 in QObject::timerEvent(QTimerEvent*) ()
   from C:\msys2\mingw32\share\qt5\plugins\platforms\qwindows.dll
#22 0x6df68460 in QObject::eventFilter(QObject*, QEvent*) ()
   from C:\msys2\mingw32\share\qt5\plugins\platforms\qwindows.dll
#23 0x6df68440 in QObject::disconnectNotify(QMetaMethod const&) ()
   from C:\msys2\mingw32\share\qt5\plugins\platforms\qwindows.dll
#24 0x6df68438 in QObject::event(QEvent*) ()
   from C:\msys2\mingw32\share\qt5\plugins\platforms\qwindows.dll
#25 0x6df6fd00 in QOpenGLStaticContext::~QOpenGLStaticContext() ()
   from C:\msys2\mingw32\share\qt5\plugins\platforms\qwindows.dll
#26 0x6df9e540 in _fu38___ZTVN10__cxxabiv120__si_class_type_infoE ()
   from C:\msys2\mingw32\share\qt5\plugins\platforms\qwindows.dll
#27 0x6df43800 in QWindowsBackingStore::scroll(QRegion const&, int, int) ()
   from C:\msys2\mingw32\share\qt5\plugins\platforms\qwindows.dll
#28 0x6df43b70 in QWindowsBackingStore::beginPaint(QRegion const&) ()
   from C:\msys2\mingw32\share\qt5\plugins\platforms\qwindows.dll
#29 0x6df43980 in QWindowsBackingStore::~QWindowsBackingStore() ()
   from C:\msys2\mingw32\share\qt5\plugins\platforms\qwindows.dll
#30 0x6df68218 in QPlatformFontDatabase::invalidate() ()
   from C:\msys2\mingw32\share\qt5\plugins\platforms\qwindows.dll
#31 0x00000000 in ?? ()
```

Last thing I did was qtcreator compilation on my specific machine/env, but result was the same. Right now I will do the same with Qt5, but because it is time consuming and might be affecting other people as well, I decided to post this issue in hope that someone might got an answer to very simple question:
"How to fix that?"

**Comment by bialasjaroslaw** [comment]
Temporary solution is to use official QtCreator and configure it to use MSYS2 Kit, but then question is what is the point of compiling QtCreator for MSYS2 if it's not gonna work? I reported issue on the Qt forum and waiting for response .

**Comment by Alexpux** [comment]
@bialasjaroslaw it not possible to test every package and also I don’t have time to fix everything too. I’m mostly use 64 bit software, while 32 just build for repo.

**Comment by Alexpux** [comment]
@bialasjaroslaw I’m try right now 32bit QtCreator and don’t have any issues with it. So problem is with your environment

**Comment by bialasjaroslaw** [comment]
@Alexpux To be clear, I am not blaming you. You did (and still do) a lot o work to keep things working, and I really appriciate that. It would be physically impossible to test everything if it's not done by software maintainer and embedded in the build process.

EDIT:
Thank you for checking that. I will make clear installation and check it ASAP.

**Comment by Alexpux** [comment]
@bialasjaroslaw check paths in your environment PATH variable. Probably you have conflicts with other libraries from different folders

**Comment by bialasjaroslaw** [comment]
@Alexpux I cleaned PATH environment variable and even msys2 mingw32 and mingw64 execs are not able to run. It turns out that they differs from fresh install (as well as mintty.exe msys-2.0.dll and bash.exe). After replacing them I saw for fraction of a second Information that:
```
Directory \drivers\etc does not exist; exiting
If directory name is garbage you need to update your msys package
```
To sum up, after fresh install everything works. I am just wondering how it happened and what I did wrong. I mean should I update msys package once in a while or it is because I spoiled PATH variable and did not notice anything until it was too late?
